pkg/disttask/framework/handle/handle.go

```diff
@@ -267,6 +271,21 @@ func GetTargetScope() string {
 	return vardef.ServiceScope.Load()
 }
 
+// GetCloudStorageURI returns the cloud storage URI with cluster ID appended to the path.
+func GetCloudStorageURI(store kv.Storage) string {
+	cloudURI := vardef.CloudStorageURI.Load()
+	if s, ok := store.(kv.StorageWithPD); ok {
+		// When setting the cloudURI value by SQL, we already checked the effectiveness, so we don't need to check it again here.
+		u, _ := litstorage.ParseRawURL(cloudURI)
+		if len(u.Path) != 0 {
+			u.Path = filepath.Join(u.Path, strconv.FormatUint(s.GetPDClient().GetClusterID(context.TODO()), 10))
```

Comment:
```suggestion
			u.Path = filepath.Join(u.Path, strconv.FormatUint(s.GetPDClient().GetClusterID(ctx), 10))
```

Maybe use the real context?